storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/KvStoreDatabase.java

```diff
@@ -426,6 +448,8 @@ private void deleteFinalizedBlocks(final List<Pair<UInt64, Bytes32>> blocksToPru
       try (final FinalizedUpdater updater = finalizedUpdater()) {
         blocksToPrune.forEach(
             pair -> updater.deleteFinalizedBlock(pair.getLeft(), pair.getRight()));
+        // update earliest finalized block slot
+        updater.setEarliestBlockSlot(blocksToPrune.getLast().getLeft().plus(ONE));
```

Comment:
so the fun bit here would be if we're in a scenario where we delete to N, but N+1 had no block... maybe N+2, N+3... depends on network quality... but i think this will then leave us in an inconsistent state which would be bad.

Reply:
you're right, but we're only updating the slot number, wouldn't that always exists even if the block is empty?

Reply:
so if we use this slot for anything, it'll be wrong if the block is missing... whats the use-case? usually when we're looking for earliest block we're going to start returning blocks in that range...

Reply:
We look use the earliest slot when we receive requests from peers to provide blocks by range. 
https://github.com/Consensys/teku/blob/015f7bd46d07b34d5f285f5b0ba8ce63110c4ca4/networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BeaconBlocksByRangeMessageHandler.java#L169-L173

It doesn't necessarily mean we're using the earliest block at all.